pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -384,15 +421,23 @@ func (r *repository) Put(manifest *schema1.SignedManifest) error {
 			return statusErr
 		}
 
-		if _, err := client.ImageStreams(r.namespace).Create(&stream); err != nil {
+		is, err := client.ImageStreams(r.namespace).Create(&stream)
```

Comment:
Not yet. But an addition of `openshift.io/imagestreams` is planned for `3.3`. So we might actually prepare the registry in advance.

I wonder if there's a better object we can generate events for. Because with this quota in place and a denied push, we may not be able to generate the event because the image stream cannot be created.